socs/agents/scpi_psu/agent.py

```diff
@@ -125,6 +182,30 @@ def set_voltage(self, session, params=None):
 
         return True, 'Set channel {} voltage to {}'.format(params['channel'], params['volts'])
 
+    @ocs_agent.param('channel', type=int, choices=[1, 2, 3])
+    def get_current(self, session, params=None):
+        """get_current(channel)
+
+        **Task** - Measure and return the current of the power supply.
+
+        Parameters:
+            channel (int): Channel number (1, 2, or 3).
+
+        """
+        chan = params['channel']
+        with self.lock.acquire_timeout(1) as acquired:
+            if acquired:
+                data = {
+                    'timestamp': time.time(),
+                    'block_name': 'output',
+                    'data': {}
+                }
+                data['data']['Current_{}'.format(chan)] = self.psu.get_curr(chan)
+                session.data = data
```

Comment:
Same comment as above in `get_voltage()`.

Reply:
done